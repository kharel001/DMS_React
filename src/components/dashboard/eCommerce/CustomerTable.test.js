import React from 'react';
import CustomerTable, {data, createData} from './CustomerTable';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const setup = (props = {}) => {
    return shallow( < CustomerTable {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


const dummy_props={
  data:{id:"id1",
  image:"image1",
  name:"name1",
  order: "order1",
  userId:"user1"
  }
}
describe("CustomerTable Comp",()=>{
it("Snapshot",()=>{
    const wrapper= setup(dummy_props);
    expect(wrapper).toMatchSnapshot()

})
it("Props",()=>{
    const wrapper= setup({...dummy_props});
    let avatarComp= wrapper.find('[data-test="customerCell"]')
    expect(avatarComp).toHaveLength(3)
    expect(avatarComp.at(2).prop('data')).toEqual({id:3, name:'Domnic Brown', userId:'Co-Founder', image:"https://via.placeholder.com/150x150", order:'0'})
    

    

})
})