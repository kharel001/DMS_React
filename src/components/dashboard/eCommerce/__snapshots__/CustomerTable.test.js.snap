// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerTable Comp Snapshot 1`] = `ShallowWrapper {}`;
